src/document.cpp

```diff
@@ -618,6 +632,49 @@ rvoe<NoReturnValue> PdfDocument::create_catalog() {
   /Pages {} 0 R
 )",
                    pages_object);
+
+    if(!page_labels.empty()) {
+        buf += R"(  /PageLabels<<
+    /Nums [)";
+        for (const auto &page_label : page_labels) {
+            buf += std::format("{} <<", page_label.page_id);
+            if (page_label.style != CAPY_PAGE_LABEL_NUMBER_STYLE_NONE) {
+                buf += R"(
+      /S )";
+            switch (page_label.style) {
+                case CAPY_PAGE_LABEL_NUMBER_STYLE_DECIMAL:
+                    buf += "D";
+                    break;
+                case CAPY_PAGE_LABEL_NUMBER_STYLE_ROMAN_UPPER:
+                    buf += "R";
+                    break;
+                case CAPY_PAGE_LABEL_NUMBER_STYLE_ROMAN_LOWER:
+                    buf += "r";
+                    break;
+                case CAPY_PAGE_LABEL_NUMBER_STYLE_LETTER_UPPER:
+                    buf += "A";
+                    break;
+                case CAPY_PAGE_LABEL_NUMBER_STYLE_LETTER_LOWER:
+                    buf += "a";
+                    break;
+                default:
+                    break;
```

Comment:
This should index into an array instead, see for example how the `colorspace_names` array is used.